src/esp/physics/RigidScene.h

```diff
@@ -19,31 +19,58 @@ class RigidScene : public RigidBase {
   /**
    * @brief Virtual destructor for a @ref RigidScene.
    */
-  virtual ~RigidScene(){};
+  virtual ~RigidScene() {}
 
   /**
-   * @brief Initializes the @ref RigidObject or @ref RigidScene that inherits
+   * @brief Initializes the @ref RigidScene that inherits
    * from this class
-   * @param physicsAttributes The template structure defining relevant
+   * @param resMgr a reference to ResourceManager object
+   * @param handle The handle for the template structure defining relevant
    * phyiscal parameters for this object
    * @return true if initialized successfully, false otherwise.
    */
-  virtual bool initialize(
-      const assets::ResourceManager& resMgr,
-      const assets::AbstractPhysicsAttributes::ptr physicsAttributes) override;
+  bool initialize(const assets::ResourceManager& resMgr,
+                  const std::string& handle) override;
+
+  /**
+   * @brief Get the template used to initialize this object or scene.
+   *
+   * AbstractPhysicsAttributes templates are able to be changed between
+   * instances of objects.
+   * @return The initialization settings of this object instance, casted to the
+   * appropriate value.
+   */
+  const std::shared_ptr<const assets::PhysicsSceneAttributes>
+  getInitializationAttributes() const {
+    return RigidBase::getInitializationAttributes<
+        assets::PhysicsSceneAttributes>();
+  };
+  /**
+   * @brief Finalize the creation of this @ref RigidScene
+   */
+  void finalizeObject() override { finalizeObject_LibSpecifc(); }
 
  private:
   /**
    * @brief Finalize the initialization of this @ref RigidScene
-   * geometry.  This is overridden by inheriting objects
+   * geometry.  This is overridden by inheriting class specific to certain
+   * physics libraries.Necessary to support kinematic objects without any
+   * dynamics support.
    * @param resMgr Reference to resource manager, to access relevant components
    * pertaining to the scene object
    * @return true if initialized successfully, false otherwise.
    */
-  virtual bool initializationFinalize(
+  bool initialization_LibSpecific(
       CORRADE_UNUSED const assets::ResourceManager& resMgr) override {
     return true;
   }
+  /**
+   * @brief any physics-lib-specific finalization code that needs to be run
+   * after@ref RigidScene is created.  Called from finalizeObject.  Overridden
+   * by inheriting class specific to certain physics libraries. Necessary to
+   * support kinematic objects without any dynamics support.
```

Comment:
```suggestion
   * after @ref RigidScene is created.  Called from finalizeObject.  Overridden
   * by inheriting class specific to certain physics libraries.
```

Reply:
Don't we want that added info?  Some future maintainer might not understand what's going on.